watertap/property_models/activated_sludge/asm1_properties.py

```diff
@@ -282,39 +283,76 @@ def build(self):
             units=pyo.units.kmol / pyo.units.m**3,
         )
 
-    def get_material_flow_terms(b, p, j):
-        if j == "H2O":
-            return b.flow_vol * b.params.dens_mass
-        elif j == "S_ALK":
-            # Convert moles of alkalinity to mass of C assuming all is HCO3-
-            return b.flow_vol * b.alkalinity * (12 * pyo.units.kg / pyo.units.kmol)
-        else:
-            return b.flow_vol * b.conc_mass_comp[j]
-
-    def get_enthalpy_flow_terms(b, p):
-        return (
-            b.flow_vol
-            * b.params.dens_mass
-            * b.params.cp_mass
-            * (b.temperature - b.params.temperature_ref)
+        # Material and energy flow and density expressions
+        def material_flow_expression(self, j):
+            if j == "H2O":
+                return self.flow_vol * self.params.dens_mass
+            elif j == "S_ALK":
+                # Convert moles of alkalinity to mass of C assuming all is HCO3-
+                return (
+                    self.flow_vol
+                    * self.alkalinity
+                    * (12 * pyo.units.kg / pyo.units.kmol)
+                )
+            else:
+                return self.flow_vol * self.conc_mass_comp[j]
+
+        self.material_flow_expression = pyo.Expression(
+            self.component_list,
+            rule=material_flow_expression,
+            doc="Material flow terms",
         )
```

Comment:
Why switch to expressions instead of using the former methods?

Reply:
Because we need to provide scaling factors for those expressions so that the control volumes can do scaling.